### ISSUE TITLE
Random coloring of highlited texts

### ISSUE THREAD

**Issue by crystalfp** [yzhang-gh/vscode-markdown#503]
Issue Type: <b>Bug</b>

#### What is the problem?
The text marked bold, italic or bold-italic is colored apparently at random when the extension is enabled.

#### How can I reproduce it?
Put the following text in `bug.md` file.
```
alcune riflessioni in merito alla necessità di ipotizzare un **nuovo «paradigma di apprendimento»** **fondato sull’integrazione fra le nuove tecnologie e i percorsi didattici tradizionali,** in un’ottica inclusiva”.
```

Open `bug.md` 

With the extension disabled, the two bold texts are colored blue (my theme is Monokai).

Now enable the extension and restart VScode. The first text is not colored (except the asterisks that are a dim gray), the second is colored correctly.

#### Is there any error message in the console?

There is no error show.

Thanks for looking!
mario

Extension version: 2.4.2
VS Code version: Code 1.37.0 (036a6b1d3ac84e5ca96a17a44e63a87971f8fcc8, 2019-08-08T02:33:50.993Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz (8 x 2904)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.81GB (6.03GB free)|
|Process Argv|--file-uri file:///d%3A/Montessori/Libro%20-%20Tecnologia%20in%20famiglia/Tecnologia%20in%20famiglia.code-workspace|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by crystalfp** [comment]
plain theme strikes again! Who knows why it has been re-enabled and by who. Certainly not by me. Disabling it again and the problem has been solved.
Sorry for the noise
mario

**Comment by crystalfp** [comment]
One info more.
If I disable "Plain Theme" from the settings editor, in the json file the setting
`"markdown.extension.syntax.plainTheme": false` does not appear. Maybe this is the problem: the default seems to be `true`. Now I added it manually to the `settings.json` file.

**Comment by yzhang-gh** [comment]
The default should be `false` 😳. Do you happen to have changed it in the workspace `settings`?

**Comment by crystalfp** [comment]
No. No trace of the setting. As I said before, to be on the safe side, I added the `"markdown.extension.syntax.plainTheme": false` to `settings.json`.

Do I remember wrongly or there was a time in which italic-marked text was rendered in italic, bold in bold and so on? Should I open a feature request?

Thanks for your patience!
mario

**Comment by yzhang-gh** [comment]
Thank you.

I think it is already done by themes. For example,

**Dark+** (default VSCode theme)
![image](https://user-images.githubusercontent.com/7588612/62824821-ec942400-bb9a-11e9-882b-62337b505ff9.png)

**Comment by crystalfp** [comment]
OK! I switched to Dark+ and now Markdown is rendered as I like. When I will have time, I will explore other themes that render Markdown "correctly.
Thanks!
mario

**Comment by Lemmingh** [comment]
@crystalfp

> Do I remember wrongly or there was a time in which italic-marked text was rendered in italic, bold in bold and so on?

Was the **Monokai** you referred to the **VS Code built-in** Monokai theme (`vscode.theme-monokai`)?

As what **yzhang-gh** said, the styles of italic and bold Markdown element in editor are handled by themes. I examined the built-in Monokai, and I think it's the root of the problem.

Compared with **Dark+**, the Monokai theme file
`<VS Code install location>/resources/app/extensions/theme-monokai/themes/monokai-color-theme.json`
misses styles for `markup` scopes.

My solution is

> **Note**
>
> Before we start, you need to know how to develop a VS Code color theme. See [Color Theme | Visual Studio Code Extension API](https://code.visualstudio.com/api/extension-guides/color-theme#create-a-new-color-theme).

1. Create a new theme

    Just copy the folder `<VS Code install location>/resources/app/extensions/theme-monokai` into your [user extension directory](https://code.visualstudio.com/docs/editor/extension-gallery#_where-are-extensions-installed) (usually `<user profile>/.vscode/extensions`), and modify `publisher` and `name` in `package.json`.

2. Copy `<VS Code install location>/resources/app/extensions/theme-defaults/themes/dark_defaults.json` to `themes/dark_defaults.json`

3. Create a file `themes/style-supplement.json` whose content is

    ```json
    {
    	"$schema": "vscode://schemas/color-theme",
    	"name": "Supplement to Styling",
    	"include": "./dark_defaults.json",
    	"tokenColors": [
    		{
    			"scope": "markup.bold",
    			"settings": {
    				"fontStyle": "bold"
    			}
    		},
    		{
    			"scope": "markup.heading",
    			"settings": {
    				"fontStyle": "bold"
    			}
    		},
    		{
    			"scope": "markup.italic",
    			"settings": {
    				"fontStyle": "italic"
    			}
    		},
    		{
    			"scope": "markup.underline",
    			"settings": {
    				"fontStyle": "underline"
    			}
    		}
    	]
    }
    ```

4. Add the following to `themes/monokai-color-theme.json` just before the line `"type": "dark",`

    ```json
    	"$schema": "vscode://schemas/color-theme",
    	"include": "./style-supplement.json",
    ```

5. You can modify other things if you like

---

BTW, @yzhang-gh

It seems that many people have no idea of the things that are not controlled by your extension. Could you add a section to `README` to document them?

Here are what I've noticed

* this issue: the styles of italic and bold Markdown element in editor are handled by VS Code themes
* issue #437, etc: the styles of preview (`Ctrl+K V`) and HTML output (`markdown.extension.printToHtml`) are handled by VS Code and CSS attached to the Markdown file
* issue #437, #506, etc: this extension does not process front matter when generating HTML because front matter varies between Markdown dialects and there is no universal way to convert it to HMTL meta so far

Thank you.

**Comment by crystalfp** [comment]
@Lemmingh Thanks for the steps to change a theme! As soon as I have time, I will try them. I solved the immediate problem by switching the theme to Dark+.

**Comment by crystalfp** [comment]
Thanks @Lemmingh ! It works as a charm. Only one thing missing. In step 1 you have to change also "label" field.

**Comment by yzhang-gh** [comment]
@Lemmingh Thanks for sharing!

> It seems that many people have no idea of the things that are not controlled by your extension. Could you add a section to `README` to document them?

A problem may be, many people don't read the `README` at all 😂. (The `README` is already too long for them.)
For now, it is not a big problem. We can consider adding an explanation if we get much more this kind of issues.
